DEVELOPING.md

```diff
@@ -149,7 +127,10 @@ sequenceDiagram
     Note right of catalog: cataloger configuration is done based on src
 ```
 
-### Package object
+Additionally, here is a [gist of using syft as a library](https://gist.github.com/wagoodman/57ed59a6d57600c23913071b8470175b) to generate a SBOM for a docker image.
```

Comment:
It looks like a user has already left a comment on this gist mentioning it's out of date - let me try it with the latest syft library

Reply:
Yea running into the same issue they hi-lighted:
```
./main.go:16:30: undefined: source.New
./main.go:33:77: cannot use source.SquashedScope (constant "Squashed" of type source.Scope) as cataloger.Config value in argument to syft.CatalogPackages
./main.go:38:19: result.Artifacts.PackageCatalog undefined (type "github.com/anchore/syft/syft/sbom".Artifacts has no field or method PackageCatalog)
./main.go:39:19: result.Artifacts.Distro undefined (type "github.com/anchore/syft/syft/sbom".Artifacts has no field or method Distro)
./main.go:43:21: undefined: syft.Encode
./main.go:43:43: undefined: format.JSONOption
```

Reply:
This is the gist I had put together around the same time using the detection portion of the source library:
https://gist.github.com/spiffcs/3027638b7ba904d07e482a712bc00d3d

Reply:
I haven't updated the gist itself, and this is one of the main changes that will be in an upcoming PR. For the meantime I can update to show your gist until the other PR lands.